presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -105,116 +100,121 @@ private InternalTable getInternalTable(ConnectorSession connectorSession, Connec
+        return getInformationSchemaTable(session, handle.getCatalogName(), handle.getSchemaTableName(), prefixes);
     }
 
-    public InternalTable getInformationSchemaTable(Session session, String catalog, SchemaTableName table, Map<String, NullableValue> filters)
+    public InternalTable getInformationSchemaTable(Session session, String catalog, SchemaTableName table, Set<QualifiedTablePrefix> prefixes)
     {
         if (table.equals(TABLE_COLUMNS)) {
-            return buildColumns(session, catalog, filters);
+            return buildColumns(session, catalog, prefixes);
         }
         if (table.equals(TABLE_TABLES)) {
-            return buildTables(session, catalog, filters);
+            return buildTables(session, catalog, prefixes);
         }
         if (table.equals(TABLE_VIEWS)) {
-            return buildViews(session, catalog, filters);
+            return buildViews(session, prefixes);
         }
         if (table.equals(TABLE_SCHEMATA)) {
             return buildSchemata(session, catalog);
         }
         if (table.equals(TABLE_TABLE_PRIVILEGES)) {
-            return buildTablePrivileges(session, catalog, filters);
+            return buildTablePrivileges(session, catalog, prefixes);
         }
 
         throw new IllegalArgumentException(format("table does not exist: %s", table));
     }
 
-    private InternalTable buildColumns(Session session, String catalogName, Map<String, NullableValue> filters)
+    private InternalTable buildColumns(Session session, String catalogName, Set<QualifiedTablePrefix> prefixes)
     {
         InternalTable.Builder table = InternalTable.builder(informationSchemaTableColumns(TABLE_COLUMNS));
-        QualifiedTablePrefix prefix = extractQualifiedTablePrefix(catalogName, filters);
-        for (Entry<SchemaTableName, List<ColumnMetadata>> entry : listTableColumns(session, metadata, accessControl, prefix).entrySet()) {
-            SchemaTableName tableName = entry.getKey();
-            int ordinalPosition = 1;
-            for (ColumnMetadata column : entry.getValue()) {
-                if (column.isHidden()) {
-                    continue;
+        for (QualifiedTablePrefix prefix : prefixes) {
+            for (Entry<SchemaTableName, List<ColumnMetadata>> entry : listTableColumns(session, metadata, accessControl, prefix).entrySet()) {
+                SchemaTableName tableName = entry.getKey();
+                int ordinalPosition = 1;
+                for (ColumnMetadata column : entry.getValue()) {
+                    if (column.isHidden()) {
+                        continue;
+                    }
+                    table.add(
+                            catalogName,
+                            tableName.getSchemaName(),
+                            tableName.getTableName(),
+                            column.getName(),
+                            ordinalPosition,
```

Comment:
Can `listTableColumns(session, metadata, accessControl, prefix)` return columns from more than one table?
What's the meaning of `ordinalPosition` then?

Reply:
What's the meaning of `ordinalPosition` then?